src/wirecloud/platform/static/js/wirecloud/wiring/Connection.js

```diff
@@ -144,7 +144,7 @@
 
             updateEndpoint: function updateEndpoint(endpoint) {
 
-                if (!(endpoint instanceof Wirecloud.wiring.Endpoint)) {
+                if (!(endpoint instanceof Wirecloud.wiring.SourceEndpoint || endpoint instanceof Wirecloud.wiring.TargetEndpoint)) {
```

Comment:
This test is not coherent with the launched exception, also it does not make sense. Revert this change.